metadata-structure/metadata-structure.md

```diff
@@ -0,0 +1,340 @@
+1. `label` displayed to the user in the dropdown menu
+1. The `effect` key is probably the most important idea here. Every `setting` has an effect associated with it and aggregating all the effects along all the settings the user has chosen completely determines the `filters` and their initial `selections` that are shown to the user. So everything can be derived from just the `settings` that are chosen
+
+## Effects
+
+```php
+├── filterTypes[]
+│   ├── id # a
+│   ├── column_id
+│   ├── use_shape_regions
+│   ├── options[]
+│   │   ├── label
+│   │   ├── id # b
+│   │   ├── description
+├── indicators
+├── plotSettingsControl{}
+│   ├── defaultEffect # (here as well)
+│   │   ├── setFilters
+│   │   ├── ... (see effect below)
+│   ├── plotSettings[]
+│   │   ├── label
+│   │   ├── id
+│   │   ├── visible
+│   │   ├── value
+│   │   ├── options[]
+│   │   │   │   ├── id
+│   │   │   │   ├── label
+│   │   │   │   ├── effect # here
+│   │   │   │   │   ├── setFilters
+│   │   │   │   │   ├── setMultiple
+│   │   │   │   │   ├── setFilterValues
+│   │   │   │   │   ├── setHidden
+```
+
+For this section we will need a more detailed view of the effects structure
+
+```php
+├── setFilters[] # 1
+│   ├── filterId # 1.1
+│   ├── label # 1.2
+│   ├── stateFilterId # 1.3
+├── setMultiple[] # 2
+├── setFilterValues{} # 3
+├── setHidden[] # 4
+```
+
+Explanation of each effect and how the frontend aggregates this:
+1. `setFilters` is how the effect specifies what `filters` the user sees from the `filterTypes` The frontend aggregates this by replacing it. So if two `settings` set filters, the last `setFilters` in the for loop would be used when going through all the `settings`. Keys:
+    1. `filterId` is the `id` key in a `filterTypes`element (`a` in the visualisation)
+    1. `label` is the title of the `filter` displayed to the user in the UI
+    1. `stateFilterId` is the id the state uses to identify this filter. This is distinct to the `filterId` because in the bubble plot we have two indicators filters that are the same filter type (so have the same `options` and the same `column_id`) however they are distinct because one controls the color of the bubbles and one controls the size of the bubbles so while both their `filterId`s are `indicator`, the `stateFilterId`s for both are different, `colorIndicator` and `sizeIndicator`. For most cases this can be set as the same thing as the `filterId`. All other effects will now identify filters by `stateFilterId` as these uniquely correspond to what the user sees
+1. `setMultiple` is an effect that sets a `filter` to a multi-select dropdown (they are all single select by default). It is an array of `stateFilterId`s (`1.3` in the visualisation) This is aggregated via concatenation in the frontend so if two settings have `setMultiple` properties `["state_id_1", "state_id_2"]` and `["state_id_3"]` respectively then all three filters with those state ids will be set to multi-selects
```

Comment:
What visualisation? Edit: Oh I see, the schema above, I was looking for an image.

Reply:
Slightly clearer to say "converts a `filter`" rather than "sets a `filter`" because I was trying to interpret the sentence like "sets" meant "applies".